source/components/triggering-changes-with-actions.md

```diff
@@ -156,3 +189,193 @@ a function instead.
 Actions in components allow you to
 decouple an event happening from how it's handled, leading to modular,
 more reusable components.
+
+## Handling Action Completion
+
+Often actions perform asynchronous tasks, such as making an ajax request to a server.
+Since actions are functions that can be passed in by a parent component, they are able to return values when called.
+The most common scenario is for an action to return a promise so that the component can handle the action's completion.
+
+In our user `button-with-confirmation` component we want to leave the confirmation modal open until we know that the
+operation has completed successfully.
+This is accomplished by expecting a promise to be returned from `onConfirm`.
+Upon resolution of the promise, we set a property used to indicate the visibility of the confirmation modal.
+
+```app/components/button-with-confirmation.js
+export default Ember.Component.extend({
+
+  actions: {
+    launchConfirmDialog() {
+      this.set('confirmShown', true);
+    },
+
+    submitConfirm() {
+      //call onConfirm with the value of the input field as an argument
+      const promise = this.get('onConfirm')()
+      promise.then(() => {
+        this.set('confirmShown', false);
+      });
+    },
+
+    cancelConfirm() {
+      this.set('confirmShown', false);
+    }
+  }
+});
+```
+
+## Passing Arguments
+
+Actions passed to a component may be invoked with arguments.
+In the `send-message` component, we want to provide a message type to send to the action.
+
+```app/templates/components/send-message.hbs
+{{button-with-confirmation text='Click to send your message.' onConfirm=(action 'sendMessage' 'info')}}
+```
+
+In this case, the code in `button-with-confirmation` does not change.
+It will still invoke `onConfirm` with no arguments.
+The action helper will add the arguments provided in the template to the call.
+
+Action arguments curry, meaning that you can provide partial arguments to the action helper and provide the rest of the
+arguments when you call the function within the component javascript file.
+For example, our `button-with-confirmation` component will now
+[yield](https://guides.emberjs.com/v2.2.0/components/wrapping-content-in-a-component/) the content of the confirmation
+dialog to collect extra information to be sent along with the `onConfirm` action:
+
+```app/templates/components/button-with-confirmation.hbs
+<button {{action 'launchConfirmDialog'}}>{{text}}</button>
+{{#if confirmShown}}
+<div class="confirm-dialog">
```

Comment:
should these be indented so they match the style elsewhere? For example: https://guides.emberjs.com/v2.2.0/templates/conditionals/